SonarQube.Common/AnalysisProperties/EnvScannerPropertiesProvider.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SonarQube.Common
+{
+    /// <summary>
+    /// Provides properties from the environment
+    /// </summary>
+    public class EnvScannerPropertiesProvider : IAnalysisPropertyProvider
+    {
+        private const string ENV_VAR_KEY = "SONARQUBE_SCANNER_PARAMS";
+        private readonly IEnumerable<Property> properties;
+
+        public static bool TryCreateProvider(ILogger logger, out IAnalysisPropertyProvider provider)
+        {
+            if (logger == null)
+            {
+                throw new ArgumentNullException("logger");
+            }
+
+            provider = null;
+            try
+            {
+                provider = new EnvScannerPropertiesProvider(Environment.GetEnvironmentVariable(ENV_VAR_KEY));
+                return true;
+            } catch (Exception e)
+            {
+                logger.LogError(Resources.ERROR_FailedParsePropertiesEnvVar, ENV_VAR_KEY);
+            }
+            return false;
+        }
+
+        public EnvScannerPropertiesProvider(string json)
+        {
+            if (json == null)
+            {
+                properties = new List<Property>();
+            }
+            else
+            {
+                properties = parseVar(json);
+            }
+        }
+
+        public IEnumerable<Property> GetAllProperties()
+        {
+            return properties;
+        }
+
+        public bool TryGetProperty(string key, out Property property)
+        {
+            return Property.TryGetProperty(key, properties, out property);
+        }
+
+        private IEnumerable<Property> parseVar(String json)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename method 'parseVar' to match camel case naming rules, consider using 'ParseVar'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS100)